### PR TITLE
Automate release by comment notifications

### PR DESCRIPTION
**Pull request by nicoddemus**
This PR introduces preparing releases by creating a new issue or commenting on existing issues.

The comment is in the format:

```
@pytestbot please prepare release from BRANCH
```

Where `BRANCH` can be `master` or one of the maintenance branches.

Here are two examples of issues that generating appropriate PRs:

* https://github.com/nicoddemus/pytest/issues/31
* https://github.com/nicoddemus/pytest/issues/32

### CODE REVIEW

File: .github/workflows/release-on-comment.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# part of our release process, see `release-on-comment.py`
+name: release on comment
+
+on:
+  issues:
+    types: [opened, edited]
+  issue_comment:
+    types: [created, edited]
```
Comment (blueyed) [inline .github/workflows/release-on-comment.yml]: Does this mean the job is run for any issue/comment being created?
Can this be limited to e.g. only for when pytestbot is mentioned?

Comment (nicoddemus) [reply .github/workflows/release-on-comment.yml]: Yep, managed to find a way so we skip right at the beginning, before even starting up a VM. 👍

File: .github/workflows/release-on-comment.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    if: (github.event.comment && contains(github.event.comment.body, '@pytestbot please')) || (github.event.issue && contains(github.event.issue.body, '@pytestbot please'))
```
Comment (nicoddemus) [inline .github/workflows/release-on-comment.yml]: > Can this be limited to e.g. only for when pytestbot is mentioned?

👍

**Comment by blueyed** [comment]
@nicoddemus 
I'm -1 on this: it is rather complex, adding overhead, and not really needed AFAICT.
But feel free to merge it / use it, given that you are doing releases mostly.. :+1:

**Comment by nicoddemus** [comment]
Well, I like the automation, plus the previous method is still valid.

**Pull request merged**